openff/system/components/foyer.py

```diff
@@ -0,0 +1,248 @@
+from abc import abstractmethod
+from copy import copy
+from typing import Dict, Set, Type
+
+from ele import element_from_atomic_number
+from foyer import Forcefield
+from foyer.exceptions import MissingForceError, MissingParametersError
+from foyer.topology_graph import TopologyGraph
+from openff.toolkit.topology import Topology
+
+from openff.system import unit as u
+from openff.system.components.potentials import Potential, PotentialHandler
+from openff.system.components.system import System
+from openff.system.models import PotentialKey, TopologyKey
+
+# Is this the safest way to achieve PotentialKey id separation?
+POTENTIAL_KEY_SEPARATOR = "-"
```

Comment:
Perhaps `PotentialKey.id` could be `Union[str, Tuple[str]]` in order to cover this use case without smooshing strings together